library/src/main/java/bio/terra/landingzone/stairway/common/utils/StairwayLoggingHook.java

```diff
@@ -0,0 +1,74 @@
+package bio.terra.landingzone.stairway.common.utils;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.HookAction;
+import bio.terra.stairway.StairwayHook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/** A {@link StairwayHook} which supplements logging at notable flight state transitions. */
+@Component
+public class StairwayLoggingHook implements StairwayHook {
+  private static final Logger logger = LoggerFactory.getLogger(StairwayLoggingHook.class);
+  private static final String FLIGHT_LOG_FORMAT = "Operation: {}, flightClass: {}, flightId: {}";
+  private static final String STEP_LOG_FORMAT =
+      "Operation: {}, flightClass: {}, flightId: {}, stepClass: {}, stepIndex: {}, direction: {}";
+
+  /** The key to use in {@link FlightMap} for storing the MDC context. */
+  public static final String MDC_FLIGHT_MAP_KEY = "mdcKey";
+
+  @Override
+  public HookAction startStep(FlightContext flightContext) {
+    logger.info(
```

Comment:
Am I understanding this correctly that all explicit MDC interactions are in Stairway code now?

Reply:
Yes, you are correct.  I should have included a link to the Stairway PR that made this change: https://github.com/DataBiosphere/stairway/pull/125

LZS now also gets more context than it previously had.  Previously, it only logged the context of the calling thread for logs emitted during the execution of each step.  Now, it…
- logs the context of the calling thread, along with generic flight context (flight class and flight ID) for the duration of the flight's execution on its new thread
- additionally logs generic step context (step class, index, and direction) for the duration of each step's execution

This matches what's been live in WSM for awhile.

Reply:
I expanded the PR description to include this info!